### PR TITLE
ISPN-10898 Add BlockHound to test suite to make sure cpu threads are n…

### PR DESCRIPTION
**Pull request by wburns**
…ever blocked

* Add BlockHound dep to javadoc

https://issues.redhat.com/browse/ISPN-10898

### CODE REVIEW

File: javadoc/javadoc-all/pom.xml [hunk 1]
```diff
@@ -235,6 +235,11 @@
          <groupId>org.infinispan</groupId>
          <artifactId>infinispan-scripting</artifactId>
       </dependency>
+
+      <dependency>
```
Comment (pruivo) [inline javadoc/javadoc-all/pom.xml]: have you tried to exclude the package?
See https://github.com/infinispan/infinispan/blob/master/javadoc/javadoc-all/pom.xml#L311

it seems a better fix than add it as a dependency. AFAIK, it is only a test dependency, right?

**Comment by wburns** [comment]
Unfortunately it is not just a test dependency. Is there a way to exclude a
class from javadoc instead. I will have to look myself later.

File: javadoc/javadoc-all/pom.xml (hunk 1 above)
Comment (danberindei) [reply javadoc/javadoc-all/pom.xml]: @pruivo ideally our users would also use BlockHound and our integration would warn them when they use blocking methods in callbacks that should be non-blocking, like listeners, or even `cache.putAsync(key, value).thenAccept(__ -> cache.put(key2, value2)`

Comment (wburns) [reply javadoc/javadoc-all/pom.xml]: Actually the more I think about it we may want to include the BlockHoundIntegration in the Javadoc so users can be aware of it. Thus I would say to keep this PR as is.

**Comment by danberindei** [comment]
Merged, thanks Will!

**Pull request merged**